source/_posts/2015-11-16-immutable-strings-in-ruby-2-dot-3.markdown

```diff
@@ -0,0 +1,136 @@
+---
+layout: post
+title: "Immutable strings in Ruby 2.3"
+category:
+date: 2015-11-16 08:39:25 -0300
+comments: true
+author:
+  name: Alexis Mas
+  email: alexis@wyeworks.com
+  twitter_handle: _axxiss
+  github_handle: axxiss
+  image: /images/team/
+  description: Software Engineer. Turning coffee into software since 2008.
+---
+
+String literals immutability is one of the new features you can find in [Ruby 2.3][1].
+But what does this implies?
+
+* Equality comparison affected
+* Better performance
+* Thread safety
+
+
+Currently strings immutability is included as opt-in, as the first step of a multi
+stage rollout and if it works well it will be [enabled by default on Ruby 3.0][2].
+So it still can end up being just an experiment. Time will tell.
+
+<!--more-->
+
+# Immutability
+
+Quoting [wikipedia][3]:
+
+> An immutable object is an object whose state cannot be modified after it is created.
+
+As immutable objects state cannot be changed it doesn't make sense to copy the
+same object over and over. Following that approach it will have three major
+consequences:
+
+* Data duplication
+* Unnecessary object allocation
+* Thread unsafe
+
+All of this can be mitigated using references to objects. Allowing the interpreter
+to copy a reference to an object instead copying the object itself. This can have
+certain impact on performance when working with a large codebase.
+
+
+# Moving towards immutability
+
+On this initial phase of Ruby's strings immutability, we can enable this feature
+for the whole project through a flag or on a per file basis using a pragma. Furthermore
+for a more detailed error stack while debugging another flag was added.
+
+If you are working on an existent project and want to introduce this new feature,
+obviously the most appropriate method is to use the pragma approach. Just add the
+following comment at the beginning of each file you want to switch to immutable
+strings and your done. Well, not really. Probably you will have some errors to fix.
+
+    # frozen_string_literal: true
+
+But, if today you woken up feeling like a kamikaze enable frozen strings for the
```

Comment:
`if today you woken up feeling like a kamikaze enable ...` -> `if you woke up today feeling like a kamikaze, enable ...`